types.go

```diff
@@ -133,6 +146,50 @@ func (source *Source) MetadataWithAdditionalTags(tags []string) []MetadataField
 	}
 }
 
+func (source *Source) AuthenticateToECR() bool {
+	logrus.Warnln("ECR integration is experimental and untested")
+	mySession := session.Must(session.NewSession(&aws.Config{
+		Region:      aws.String(source.AwsRegion),
+		Credentials: credentials.NewStaticCredentials(source.AwsAccessKeyId, source.AwsSecretAccessKey, ""),
+	}))
+	client := ecr.New(mySession)
+	// If a role arn has been supplied, then assume role and get a new session
+	if source.AwsRoleArn != "" {
```

Comment:
looks like we can get a simple refactor here, so we don't have to create a New client twice if the `AwsRoleArn` is defined. We are thinking of something like:
```
	var config aws.Config

	// If a role arn has been supplied, then assume role
	if source.AwsRoleArn != "" {
		config = aws.Config{Credentials: stscreds.NewCredentials(mySession, source.AwsRoleArn)}
	}

	client := ecr.New(mySession, &config)
```

Still not sure if it works as we are unable to test it locally.

Reply:
I can't see any reason why that wouldn't work. I'm busy on other stuff at the moment, but Friday might see me with a chance to fix up the conflicts and address this comment as well as the minor doc updates that I've somehow missed. Thanks for the review @YoussB!

Reply:
Thanks @mattburgess